### PR TITLE
More informative errors, and a touch of python3 compatibility.

### PR DESCRIPTION
**Pull request by cmcneil**

### CODE REVIEW

File: cottoncandy/interfaces.py [hunk 1]
```diff
@@ -46,6 +46,12 @@
                     DEFAULT_ACL, MPU_CHUNKSIZE, MPU_THRESHOLD, DASK_CHUNKSIZE, MB, SEPARATOR,
                     MAGIC_CHECK)
 
+# ------------------
+# Exceptions
+# ------------------
+
+class S3ObjectNotFoundException(Exception):
```
Comment (anwarnunez) [inline cottoncandy/interfaces.py]: This might be better placed in the S3 interface.

File: cottoncandy/interfaces.py [hunk 2]
```diff
@@ -410,7 +416,8 @@ def download_json(self, object_name):
         json_data : dict
             Dictionary representation of JSON file
         """
-        assert self.exists_object(object_name)
+        if not self.exists_object(object_name):
```
Comment (anwarnunez) [inline cottoncandy/interfaces.py]: For all of these, maybe the S3/GDrive interface should return the appropriate error. i.e. `S3ObjectNotFound` or `GDriveObjectNotFound`

**Comment by cmcneil** [comment]
Do you know what's up with this test failing? It doesn't look like it has anything to do with my changes.

**Comment by anwarnunez** [comment]
Unfortunately, Travis doesn't give access to encrypted environmental variables to repo forks. The encrypted environmental variables contain the necessary AWS credentials to run the round trip tests.

File: cottoncandy/interfaces.py (hunk 2 above)
Comment (cmcneil) [inline cottoncandy/interfaces.py]: Actually, I think I'll change it to just use FileNotFoundError. There's no reason for there to be a difference in the two backends for what I'm doing right now.

Comment (anwarnunez) [inline cottoncandy/interfaces.py]: Thought about this more. It'd be cleaner to add a function `assert_object_exists()` in `utils`, then just use that in all these places.

**Comment by cmcneil** [comment]
Suggestion accepted, but instead of making a new function, I decided it made more sense to wrap this functionality inside the exists_object() method, using an optional param. You would never need the new function in a context that doesn't have that one.

**Comment by anwarnunez** [comment]
kewl

**Pull request merged**